### ISSUE TITLE
[plug-in]Support VS Code XML Tooling

### ISSUE THREAD

**Issue by apupier** [theia-ide/theia#3317]
currently it is failing with:

```
root ERROR [hosted-plugin: 211] (node:211) UnhandledPromiseRejectionWarning: SyntaxError: Unexpected token u in JSON at position 0
    at JSON.parse (<anonymous>)
    at getSettings (/tmp/vscode-unpacked/vscode_extension_redhat_vscode_xml.vsix/extension/out/src/extension.js:74:29)
    at requirements.resolveRequirements.catch.then.requirements (/tmp/vscode-unpacked/vscode_extension_redhat_vscode_xml.vsix/extension/out/src/extension.js:45:48)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:189:7)
```

the corresponding code is:
```
function getSettings() {
        let configXML = vscode_1.workspace.getConfiguration();
        configXML = configXML.get('xml');
        let x = JSON.stringify(configXML);
        let settings = JSON.parse(x);
        settings['logs']['file'] = logfile;
        return settings;
    }
```

Line 74 is the JSON.parse

**Comment by apupier** [comment]
//cc @benoitf

**Comment by apupier** [comment]
seems that it is now another error: https://github.com/eclipse/che/issues/12536

**Comment by vrubezhny** [comment]
> seems that it is now another error: [eclipse/che#12536](https://github.com/eclipse/che/issues/12536)

This error is going to be addressed in https://github.com/theia-ide/theia/issues/3540, I've tested the fix referenced from #3540 https://github.com/theia-ide/theia/commit/549668d17371553b3dc0335da2bb0131e67e641b and it looks like it does the thing: I was able to deploy `vscode:extension/redhat.vscode-xml` with no `TypeError: Cannot read property 'editor.autoClosingBrackets' of undefined` error appeared. However there is no PR made yet for #3540 - it is still in progress.

**Comment by vrubezhny** [comment]
After the PR for https://github.com/theia-ide/theia/issues/3540 is merged, VSCode XML extension plug-in can be successfully deployed into theia and works fine.
I believe we can close this issue.
